### PR TITLE
Update the apps module namespace

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #4459 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: examples/example_plugin/example_plugin/jobs.py [hunk 1]
```diff
@@ -4,13 +4,18 @@
...
+from nautobot.apps.jobs import (
+    DryRunVar,
+    IntegerVar,
+    Job,
+    JobButtonReceiver,
+    JobHookReceiver,
+    register_jobs,
+)
 from nautobot.dcim.models import Device, Location
 from nautobot.extras.choices import ObjectChangeActionChoices
```
Comment (glennmatthews) [inline examples/example_plugin/example_plugin/jobs.py:16]: Should `ObjectChangeActionChoices` be added to apps namespace?

File: nautobot/apps/__init__.py [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 from nautobot.core.signals import nautobot_database_ready
 from nautobot.core.settings_funcs import ConstanceConfigItem
 from nautobot.extras.plugins import NautobotAppConfig
 
+
 __all__ = (
     "ConstanceConfigItem",
     "nautobot_database_ready",
     "NautobotAppConfig",
+    "NautobotConfig",
```
Comment (glennmatthews) [inline nautobot/apps/__init__.py:13]: I don't think `NautobotConfig` should be included - apps should always be using its subclass `NautobotAppConfig` that's already here.

File: nautobot/apps/api.py [hunk 3]
```diff
@@ -9,26 +9,100 @@
...
+from nautobot.core.api.fields import (
+    ChoiceField,
+    ContentTypeField,
+    NautobotHyperlinkedRelatedField,
+    ObjectTypeField,
+    SerializedPKRelatedField,
+    TimeZoneSerializerField,
+)
+from nautobot.core.api.mixins import WritableSerializerMixin
+from nautobot.core.api.pagination import OptionalLimitOffsetPagination
```
Comment (glennmatthews) [inline nautobot/apps/api.py:21]: Not sure this is needed - it should be part of the default configuration of Nautobot, not something apps should ever need to relate to directly.

File: nautobot/apps/api.py [hunk 4]
```diff
@@ -9,26 +9,100 @@
...
+from nautobot.core.api.parsers import NautobotCSVParser
+from nautobot.core.api.renderers import FormlessBrowsableAPIRenderer, NautobotCSVRenderer, NautobotJSONRenderer
```
Comment (glennmatthews) [inline nautobot/apps/api.py:23]: Also not sure these are needed by apps - they're infra. I can't immediately picture a case where an app would need to access any of these.

File: nautobot/apps/api.py [hunk 5]
```diff
@@ -9,26 +9,100 @@
...
+    get_api_version_serializer,
+    get_nested_serializer_depth,
+    get_relation_info_for_nested_serializers,
+    get_serializer_for_model,
+    get_view_name,
+    is_api_request,
+    nested_serializer_factory,
+    nested_serializers_for_models,
+    rest_api_server_error,
+    return_nested_serializer_data_based_on_depth,
```
Comment (glennmatthews) [inline nautobot/apps/api.py:42]: Also hesitant about some of the `nested` stuff, do these need to be called directly by apps at any point or are they part of the internal magic that apps don't need to care about?

File: nautobot/apps/choices.py [hunk 6]
```diff
@@ -1,5 +1,21 @@
 """Utilities for implementing choices."""
 
-from nautobot.core.choices import ChoiceSet
+from nautobot.core.choices import (
+    ButtonActionColorChoices,
+    ButtonActionIconChoices,
+    ChoiceSet,
+    ChoiceSetMeta,
```
Comment (glennmatthews) [inline nautobot/apps/choices.py:7]: Do apps need `ChoiceSetMeta`?

File: nautobot/apps/filters.py [hunk 7]
```diff
@@ -1,12 +1,54 @@
...
+    CustomFieldDateFilter,
+    CustomFieldFilterMixin,
+    CustomFieldJSONFilter,
+    CustomFieldMultiSelectFilter,
+    CustomFieldMultiValueCharFilter,
+    CustomFieldMultiValueDateFilter,
+    CustomFieldMultiValueNumberFilter,
+    CustomFieldMultiValueSelectFilter,
+    CustomFieldNumberFilter,
+)
```
Comment (glennmatthews) [inline nautobot/apps/filters.py:43]: Do apps need these?

File: nautobot/apps/graphql.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    generate_custom_field_resolver,
+    generate_filter_resolver,
+    generate_list_resolver,
+    generate_list_search_parameters,
+    generate_null_choices_resolver,
+    generate_relationship_resolver,
+    generate_restricted_queryset,
+    generate_schema_type,
+    generate_single_item_resolver,
+)
```
Comment (glennmatthews) [inline nautobot/apps/graphql.py:16]: I would have thought all of these are internals.

File: nautobot/apps/graphql.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    extend_schema_type,
+    extend_schema_type_computed_field,
+    extend_schema_type_config_context,
+    extend_schema_type_custom_field,
+    extend_schema_type_filter,
+    extend_schema_type_null_field_choice,
+    extend_schema_type_relationships,
+    extend_schema_type_tags,
+    generate_query_mixin,
+)
```
Comment (glennmatthews) [inline nautobot/apps/graphql.py:27]: These too.

File: nautobot/apps/choices.py (hunk 6 above)
Comment (gsnider2195) [reply nautobot/apps/choices.py:7]: It doesn't appear to be used by any app in the `nautobot` org

File: nautobot/apps/api.py (hunk 5 above)
Comment (gsnider2195) [reply nautobot/apps/api.py:42]: In 2.0 I don't see why the apps would care to mess with nesting serializers. FWIW I don't see any of these being imported in the apps in the `nautobot` org

File: nautobot/apps/api.py (hunk 4 above)
Comment (gsnider2195) [reply nautobot/apps/api.py:23]: Agreed. Not used anywhere by the `nautobot` org apps.

File: nautobot/apps/filters.py (hunk 7 above)
Comment (gsnider2195) [reply nautobot/apps/filters.py:43]: Doesn't look like they use any of these

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**